Virt: Update DPDK cluster configuration instructions (#59891)
openshift/openshift-docs: modules/virt-configuring-cluster-dpdk.adoc

  name: profile-1
spec:
  cpu:
    isolated: 4-39,44-79
    reserved: 0-3,40-43
  hugepages:
    defaultHugepagesSize: 1G
    pages:
    - count: 32
      node: 0
      size: 1G
  net:
    userLevelNetworking: true
  nodeSelector:
    node-role.kubernetes.io/worker-dpdk: ""
  numa:
    topologyPolicy: single-numa-node
----
+
[NOTE]
====
The compute nodes automatically restart after you apply the `MachineConfigPool` and `PerformanceProfile` manifests.
====

> . Get the generated runtimeClass from the PerformanceProfile's status:

Reviewer: ```suggestion
. Retrieve the name of the generated `RuntimeClass` resource from the `status.runtimeClass` field of the `PerformanceProfile` object :
```

Author: Done.

> . Get the generated runtimeClass from the PerformanceProfile's status:

Reviewer: ```suggestion
. Retrieve the name of the generated `RuntimeClass` resource from the `status.runtimeClass` field of the `PerformanceProfile` object :
```

Author: This seems like a duplicate of the previous comment.

> [source,terminal]

Reviewer: Fix indentation

```suggestion
+
[source,terminal]
```

Author: Done.

----
$ oc get performanceprofiles.performance.openshift.io profile-1 -o=jsonpath='{.status.runtimeClass}{"\n"}'
----

> . Set the container runtime configuration for the `virt-launcher` pod by adding the following annotation to the `HyperConverged` custom resource (CR):

Reviewer: ```suggestion
. Set the container runtime configuration for the `virt-launcher` pod, specifying the`RuntimeClass` name:
```

Reviewer: I realize that you only moved this step; my suggestion was more to tie it to the previous step and highlight that the `RuntimeClass` name was a user-supplied value.

Reviewer 2: What about merging the two with:

```
Set the earlier obtained `RuntimeClass` name as the default container runtime class for the `virt-launcher` pods by adding the following annotation to the `HyperConverged` custom resource (CR):
```

Author: Changed the wording according to @phoracek's suggestion.

+
[source,terminal]
----
$ oc annotate --overwrite -n openshift-cnv hco kubevirt-hyperconverged \

>     kubevirt.kubevirt.io/jsonpatch='[{"op": "add", "path": "/spec/configuration/defaultRuntimeClass", "value": <runtimeclass-name>}]'

Reviewer: ```suggestion
    kubevirt.kubevirt.io/jsonpatch='[{"op": "add", "path": "/spec/configuration/defaultRuntimeClass", "value": <runtimeclass_name>}]'
```

  name: profile-1
spec:
  cpu:
    isolated: 4-39,44-79
    reserved: 0-3,40-43
  hugepages:
    defaultHugepagesSize: 1G
    pages:
    - count: 32
      node: 0
      size: 1G
  net:
    userLevelNetworking: true
  nodeSelector:
    node-role.kubernetes.io/worker-dpdk: ""
  numa:
    topologyPolicy: single-numa-node
----
+
[NOTE]
====
The compute nodes automatically restart after you apply the `MachineConfigPool` and `PerformanceProfile` manifests.
====

. Retrieve the name of the generated `RuntimeClass` resource from the `status.runtimeClass` field of the `PerformanceProfile` object:
+
[source,terminal]
----
$ oc get performanceprofiles.performance.openshift.io profile-1 -o=jsonpath='{.status.runtimeClass}{"\n"}'
----

. Set the container runtime configuration for the `virt-launcher` pod by adding the following annotation to the `HyperConverged` custom resource (CR):
+
[source,terminal]
----
$ oc annotate --overwrite -n openshift-cnv hco kubevirt-hyperconverged \
    kubevirt.kubevirt.io/jsonpatch='[{"op": "add", "path": "/spec/configuration/defaultRuntimeClass", "value": <runtimeclass_name>}]'
----
+
[NOTE]
====

> Adding the annotation to the `HyperConverged` CR changes a global setting that affects all VMs that are created after the annotation is applied.

Reviewer 2: ```suggestion
Adding the annotation to the `HyperConverged` CR changes a global setting that affects all VMs that are created after the annotation is applied. Setting this annotation breaches support of given OpenShift Virtualization instance and should be used only on test clusters. For best performance, apply for a support exception.
```

Author: Done.
